packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useCallback, SetStateAction, useRef } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import { injectGlobal } from '@leafygreen-ui/emotion';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  XXSmall: 'xxsmall',
+  XSmall: 'xsmall',
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+  XLarge: 'xlarge',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+const defaultSpacing = 18;
+
+const mainContainer = css`
+  animation: fade-in 250ms ease-in-out;
+  -webkit-animation: fade-in 250ms ease-in-out;
```

Comment:
I think we can remove this and Emotion will do this for us.

Reply:
👍